collectors/cgroups.plugin/sys_fs_cgroup.c

```diff
@@ -908,12 +908,14 @@ static inline void cgroup_get_chart_name(struct cgroup *cg) {
                 if(likely(!name_error))
                     cg->pending_renames = 0;
 
-                freez(cg->chart_title);
-                cg->chart_title = cgroup_title_strdupz(s);
+                if(likely(cg->pending_renames < 2)) {
```

Comment:
What does this if achieve? I don't understand it.

Reply:
Essentially, the first time we call rename, right after setting pending_renames to 2, we through away the value if we got the error. The second time, we keep it. Got it.

Reply:
There was a problem with a docker container renaming.  `cg->chart_id = cgroup_chart_id_strdupz(s)` will change the name of the docker container on the first rename and renaming will fail on the second attempt.